src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2126,21 +2118,15 @@ public void run()
                     if (logger.isTraceEnabled())
                         logger.trace("Async instance tidier for {}, after barrier", descriptor);
 
-                    if (bf != null)
-                        bf.close();
-                    if (summary != null)
-                        summary.close();
                     if (runOnClose != null)
                         runOnClose.run();
-                    if (dfile != null)
-                        dfile.close();
-                    if (ifile != null)
-                        ifile.close();
+                Throwable ex = Throwables.close(null, closables);
```

Comment:
Since here we are changing this to continue closing files after an exception, can we also fix the method to stash exceptions and continue for all steps of the process after the barrier?

In particular, I believe we should still release the global ref even on error during closing, and it also makes sense to report the errors here (and send them to `JVMStabilityInspector.inspectThrowable`) instead of passing them on to whatever catches exceptions in `nonPeriodicTasks`.

Nit: indent appears wrong.

Reply:
ok, I'll fix that